globalsat_sport.cc

```diff
@@ -0,0 +1,742 @@
+			trk = route_head_alloc();
+			trk->rte_name = xstrdup(str);
+			trk->rte_desc = xstrdup("GH625XT GPS tracklog data");
+
+			track_add_head (trk);
+
+			uint8_t GetTrack[5];
+			GetTrack[0] = CommandGetTrackFileSections;
+			GetTrack[1] = 0x0;
+			GetTrack[2] = 0x1;
+			GetTrack[3] = (0xFF00 & be32toh (header->gh_ptrec.Index)) >> 8;
+			GetTrack[4] = 0xFF & be32toh (header->gh_ptrec.Index);
+			globalsat_write_package (GetTrack, 5);
+
+			uint8_t trackDeviceCommand;
+			int track_length;
+			uint8_t *track_payload = NULL;
+			track_payload = globalsat_read_package (&track_length, &trackDeviceCommand);
+			if ((track_length == 0) || (track_payload == NULL)) {
+				//TODO error
+			}
+			//      printf ("Got track package!!! Train data\n");
+
+			gh_db_train *db_train = (gh_db_train *) track_payload;
+			if (global_opts.debug_level > 1) {
+				printf ("\nTrainData:%02d-%02d-%02d ", db_train->dateStart.Year,db_train->dateStart.Month, db_train->dateStart.Day);
+				printf ("%02d:%02d:%02d ", db_train->timeStart.Hour, db_train->timeStart.Minute, db_train->timeStart.Second);
+				printf ("Total(points:%6d time:%6ds dist:%9dm) LapCnts:%5d ", be32toh (db_train->TotalPoint),be32toh (db_train->TotalTime) / 10,be32toh (db_train->TotalDistance), be16toh (db_train->LapCnts));
+				printf ("Index/StartPt:%d ", be32toh (db_train->gh_ptrec.Index));
+				printf ("LapIndex/EndPt:%d ", be32toh (db_train->gh_laprec.LapIndex));
+				printf ("MultiSport:0x%x ", db_train->MultiSport);
+			}
+			int total_laps = be16toh (db_train->LapCnts);
+			int total_laps_left = total_laps;
+			free (track_payload);
+			track_payload = NULL;
+
+
+			// Get laps
+			while (total_laps_left > 0) {
+				globalsat_send_simple (CommandGetNextTrackSection);
+				track_payload = globalsat_read_package (&track_length, &trackDeviceCommand);
+				if ((track_length == 0) || (track_payload == NULL)) {
+					//TODO error
+				}
+				//	printf ("Got track package!!! Laps data\n");
+
+				gh_trainheader *header = (gh_trainheader *) track_payload;
+				if (global_opts.debug_level > 1) {
+					printf ("Lap Trainheader: %02d-%02d-%02d ", header->dateStart.Year, header->dateStart.Month, header->dateStart.Day);
+					printf ("%02d:%02d:%02d ", header->timeStart.Hour, header->timeStart.Minute, header->timeStart.Second);
+					printf ("Total(points:%6d time:%6ds dist:%9dm) LapCnts:%5d ", be32toh (header->TotalPoint),be32toh (header->TotalTime) / 10, be32toh (header->TotalDistance), be16toh (header->LapCnts));
+					printf ("Index/StartPt:%d ", be32toh (header->gh_ptrec.Index));
+					printf ("LapIndex/EndPt:%d ", be32toh (header->gh_laprec.LapIndex));
+					printf ("DataType:0x%x\n", header->DataType);
+				}
+				/*
+				 * GPS year: 2000+; struct tm year: 1900+
+				 * GPS month: 1-12, struct tm month: 0-11
+				 */
+/*				gpstime.tm_year  = header->dateStart.Year+100;
```

Comment:
Heh. Looks like you started this before we had the MUCH more sensible QDateTime family.